test/issues/pull.ts

```diff
@@ -0,0 +1,102 @@
+import * as Iridium from "../../iridium";
+import * as chai from "chai";
+
+interface EventType
+{
+    type: string;
+    time: Date;
+};
+
+interface TestDocument {
+    _id?: string;
+
+    stuff?: {
+        [name: string]: {
+            [tracker: number]: EventType[];
+        }
+    };
+}
+
+
+@Iridium.Collection("test")
+@Iridium.Transform(doc => {
+    if (doc.metadata && typeof doc.metadata === "string")
+        doc.metadata = JSON.parse(doc.metadata);
+    return doc;
+}, doc => {
+    if (doc.metadata && typeof doc.metadata === "object")
+        doc.metadata = JSON.stringify(doc.metadata);
+    return doc
+})
+class Test extends Iridium.Instance<TestDocument, Test> implements TestDocument {
+    @Iridium.ObjectID
+    _id: string;
+
+    @Iridium.Property({
+        $propertyType: {
+            $propertyType: [{
+                type: String,
+                time: Date,
+                position: {
+                    $type: {
+                        type: /^Point$/,
+                        coordinates: [Number, 2, 2]
+                    },
+                    $required: false
+                },
+                data: {
+                    $type: Object,
+                    $required: false
+                }
+            }]
+        }
+    })
+    stuff: {
+        [name: string]: {
+            [event: number]: EventType[];
+        }
+    };
+}
+
+describe("issues", () => {
+    let core = new Iridium.Core({ database: "test" });
+    before(() => core.connect());
+    after(() => core.close());
+
+    describe("pull", () => {
```

Comment:
Do you mind renaming this to #82 now that you've got an issue number, it'll make keeping track of what this tests a lot easier in future for me. If you could also rename the file itself, that'd be amazing.